src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -70,11 +72,16 @@ def execute(self):
     with self.invalidated(binaries, invalidate_dependents=True) as invalidation_check:
       python_deployable_archive = self.context.products.get('deployable_archives')
       python_pex_product = self.context.products.get('pex_archives')
+      precedence = python_setup.default_precedence[:]
```

Comment:
since this is `tuple` backed, there should be no need to explicitly copy via `[:]` since tuples aren't mutable.